qa/rpc-tests/test_framework/util.py

```diff
@@ -78,8 +90,16 @@ def initialize_chain(test_dir):
     bitcoind and bitcoin-cli must be in search path.
     """
 
-    if not os.path.isdir(os.path.join("cache", "node0")):
-        devnull = open("/dev/null", "w+")
+    if ( not os.path.isdir(os.path.join("cache", "node0"))
+             or not os.path.isdir(os.path.join("cache", "node1")) 
+             or not os.path.isdir(os.path.join("cache", "node2")) 
+             or not os.path.isdir(os.path.join("cache", "node3")) ):
+
+        #find and delete old cache directories if any exist
+        for i in range(4):
+	    if os.path.isdir(os.path.join("cache", "node"+str(i))): shutil.rmtree(os.path.join("cache", "node"+str(i)))
```

Comment:
So you only want to delete existing cache dirs when at least one cache dir needs to be created?